unittests/test_utility.py

```diff
@@ -19,492 +17,547 @@
+                                  timeout=3)
+    assert not os_ext.git_repo_exists('reframe.git', timeout=3)
+    assert not os_ext.git_repo_exists('https://github.com/eth-cscs/xxx',
+                                      timeout=3)
+
+
+def test_force_remove_file(tmp_path):
+    fp = tmp_path / 'tmp_file'
+    fp.touch()
+    fp_name = str(fp)
+
+    assert os.path.exists(fp_name)
+    os_ext.force_remove_file(fp_name)
+    assert not os.path.exists(fp_name)
+
+    # Try to remove a non-existent file
+    os_ext.force_remove_file(fp_name)
+
+
+def test_expandvars_dollar():
+    text = 'Hello, $(echo World)'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    # Test nested expansion
+    text = '$(echo Hello, $(echo World))'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_backticks():
+    text = 'Hello, `echo World`'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    # Test nested expansion
+    text = '`echo Hello, `echo World``'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_mixed_syntax():
+    text = '`echo Hello, $(echo World)`'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    text = '$(echo Hello, `echo World`)'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_error():
+    text = 'Hello, $(foo)'
+    with pytest.raises(SpawnedProcessError):
+        os_ext.expandvars(text)
+
+
+def test_strange_syntax():
+    text = 'Hello, $(foo`'
+    assert 'Hello, $(foo`' == os_ext.expandvars(text)
+
+    text = 'Hello, `foo)'
+    assert 'Hello, `foo)' == os_ext.expandvars(text)
+
+
+def test_expandvars_nocmd(monkeypatch):
+    monkeypatch.setenv('FOO', 'World')
+    text = 'Hello, $FOO'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    text = 'Hello, ${FOO}'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+@pytest.fixture
+def prefix_abs(tmp_path):
+    prefix = tmp_path / 'prefix'
+    prefix.mkdir()
+    return prefix.resolve()
+
+
+@pytest.fixture
+def target_abs(tmp_path):
+    target = tmp_path / 'target'
+    target.mkdir()
+    return target.resolve()
+
+
+@pytest.fixture
+def testfiles(prefix_abs, target_abs):
+    # Create a test directory structure
+    #
+    # prefix/
+    #   bar/
+    #     bar.txt
+    #     foo.txt
+    #     foobar.txt
+    #   foo/
+    #     bar.txt
+    #   bar.txt
+    #   foo.txt
+    #
+    (prefix_abs / 'bar').mkdir(parents=True)
+    (prefix_abs / 'foo').mkdir(parents=True)
+    (prefix_abs / 'bar' / 'bar.txt').touch()
+    (prefix_abs / 'bar' / 'foo.txt').touch()
+    (prefix_abs / 'bar' / 'foobar.txt').touch()
+    (prefix_abs / 'foo' / 'bar.txt').touch()
+    (prefix_abs / 'bar.txt').touch()
+    (prefix_abs / 'foo.txt').touch()
+
+    # Create also a subdirectory in target, so as to check the recursion
+    (target_abs / 'foo').mkdir(parents=True)
+
+
+@pytest.fixture()
+def verify_target_directory(prefix_abs, target_abs, testfiles):
```

Comment:
This should not be a fixture. It just asserts some things. Make it a normal function and rename it to `assert_directory_structure()`.